### PR TITLE
Phase 5: EMA Decay Sweep + Earlier EMA Start Epoch

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The EMA model provides final predictions, so EMA hyperparameters directly affect the quality of submitted results. The current config uses `ema_decay=0.999` and `ema_start_epoch=140` (starting EMA late in training). Two untested levers: (1) higher decay (0.9995, 0.9999) keeps more historical model weight and may smooth over noisy late-training updates; (2) earlier EMA start (epoch 80/100/120) gives the EMA more training signal to integrate. These are cheap changes with potentially meaningful impact on final metrics.

## Instructions

Run an 8-GPU sweep varying EMA decay and start epoch. All other hyperparameters stay at baseline:

```bash
# GPU 0 — decay=0.9995, start=140 (baseline start, higher decay)
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay9995_start140" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.9995 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 140

# GPU 1 — decay=0.9999, start=140
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay9999_start140" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.9999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 140

# GPU 2 — decay=0.999 (baseline), start=80
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay999_start80" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 80

# GPU 3 — decay=0.999 (baseline), start=100
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay999_start100" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 100

# GPU 4 — decay=0.999 (baseline), start=120
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay999_start120" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 120

# GPU 5 — decay=0.9995, start=100
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay9995_start100" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.9995 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 100

# GPU 6 — decay=0.9999, start=100
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/decay9999_start100" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.9999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 100

# GPU 7 — baseline control (decay=0.999, start=140)
python train.py --agent nezuko --wandb_group phase5/ema-optimization --wandb_name "nezuko/baseline_ctrl" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep --ema_start_epoch 140
```

Report the best run's val/loss and full surface MAE breakdown. Note which EMA config won.

## Baseline

- **val/loss**: 0.401
- **p_in**: 12.95
- **p_oodc**: 8.40
- **p_tan**: 33.8
- **p_re**: 24.7
- **Baseline reproduce command**:
```bash
python train.py --agent nezuko --wandb_name "nezuko/baseline" \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 180 --disable_pcgrad \
  --pressure_first --pressure_deep
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: ## Review — EMA Optimization

### Decision: **CLOSE** — Baseline EMA config is already near-optimal

Good sweep confirming the current defaults are well-tuned. The control (decay=0.999, start=140) achieved val/loss=0.3998, matching baseline. The start=120 variant's p_in=12.5 is interesting but within single-seed noise.

Moving you to a bold experiment per human team direction.
